test/suite/run.py

```diff
@@ -318,6 +321,21 @@ def error(exitval, prefix, msg):
             if option == '-asan':
                 asan = True
                 continue
+            if option == '-batch' or option == 'b':
+                if batchtotal != 0 or len(args) == 0:
```

Comment:
Does the checking of `batchtotal != 0` intend to capture repeated batch argument settings?

Reply:
Yes, checking `batchtotal != 0` catches multiple -b.  There are some situations where "repeating" an arg makes sense, but I don't think -b is one of them.